src/core/core.controller.js

```diff
@@ -35,16 +35,80 @@ defaults._set('global', {
 	responsiveAnimationDuration: 0
 });
 
+/**
+ * Recursively merge the given config objects as the `scales` options by
+ * incorporating scale defaults in `xAxes` and `yAxes` array items, then
+ * returns a deep copy of the result, thus doesn't alter inputs.
+ */
+function mergeScaleConfig(/* config objects ... */) {
+	return helpers.merge({}, Array.prototype.slice.call(arguments), {
+		merger: function(key, target, source, options) {
+			if (key === 'xAxes' || key === 'yAxes') {
+				var slen = source[key].length;
+				var i, type, scale;
+
+				if (!target[key]) {
+					target[key] = [];
+				}
+
+				for (i = 0; i < slen; ++i) {
+					scale = source[key][i];
+					type = helpers.valueOrDefault(scale.type, key === 'xAxes' ? 'category' : 'linear');
```

Comment:
there is a global alias for `valueOrDefault`

Reply:
done